File: TEMA 1/pb2_nrpits.py
def count_pits(sir):
    max = 1000001
    nrpits = 0
    stack = [max]

    for i in range(n):
        a = sir[i]
        while a > stack[-1]:
            stack.pop()
            if len(stack) > 1:
                nrpits = nrpits + 1
        stack.append(a)

    with open("nrpits.out", "w") as file:
        file.write(str(nrpits))


with open("nrpits.in", "r") as file:
    n = int(file.readline().strip())
    sir = [int(x) for x in file.readline().strip().split()]
    count_pits(sir)
        

File: TEMA 1/pb2_nrpits.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <stack>

void solve(const std::vector<int>& sir) {
    const int max = 1000001;
    int count_pits = 0;
    std::stack<int> stack;
    stack.push(max);

    for (int a : sir) {
        while (a > stack.top()) {
            stack.pop();
            if (stack.size() > 1) {
                count_pits++;
            }
        }
        stack.push(a);
    }

    std::ofstream outfile("nrpits.out");
    outfile << count_pits;
    outfile.close();
}

int main() {
    std::ifstream infile("nrpits.in");
    int n;
    infile >> n;

    std::vector<int> sir(n);
    for (int i = 0; i < n; ++i) {
        infile >> sir[i];
    }

    infile.close();
    solve(sir);

    return 0;
}
